tests/Unit/RedisSentinelServiceProviderTest.php

```diff
@@ -197,4 +197,22 @@ public function testBootExtendsSessionHandlers()
             $this->assertNotNull($this->app->session->driver('redis-sentinel'));
         }
     }
+
+    public function testAutoBootOption()
+    {
+        $this->app->config->set('redis-sentinel.auto-boot', false);
+        $this->provider->register();
+
+        // It didn't auto boot
+        $this->assertNull($this->app->cache->store('redis-sentinel'));
```

Comment:
Good idea, I'm getting some other test failures on my local environment (horizon related) and I missed that one. I've just pushed an update through with it.

Reply:
Pre-coffee commits are never a good idea.. The unit tests run, however my local PHP install's missing Horizon related dependencies so there's a lot of white noise and my sleepy brain didn't pick up that one exeption was mine..

I've fixed the import issue and tracked auto boot failing down to the cache cleaning - it removes the auto_boot option. I've added in an explicit check to retain it, debating whether I add in a whitelist of config options to retain to future-proof it a little.

Reply:
Agreed...coffee before code :) 

Something like this will probably work as a "whitelist" for the time being:

```php
protected function cleanPackageConfiguration()
{
    ...
    if ($this->config->get('redis-sentinel.clean_config', true) === true) {
        $this->config->set('redis-sentinel', [
            'auto_boot' => $this->config->get('redis-sentinel.auto_boot', false),
            'Config merged. Set redis-sentinel.clean_config=false to keep.',
        ]);
    }
    ...
}
```

I don't foresee adding a lot of package initialization options, and the config loader is due for a refactor...it grew enough that it's hard to work with now. 

Great work so far. Some of the Horizon tests on Travis fail occasionally when they're under higher load (the timeout parameters in the integration tests are too tight--I need to tweak those). Testing is the most frustrating part of working with this otherwise simple package--it aims to support so many use cases and framework versions at this point. Let me know if you want to keep plugging away. I can merge and finish up the tests for this feature if you prefer.

Reply:
I've just merged that in, much cleaner than my original code. The test suite is impressively comprehensive, I normally only have time to cover the bare minimum.

It looks like there's issues with the Laravel 5.7 tests, although they're passing locally and it looks unrelated to this commit, I won't have time to check why until tomorrow though.

Reply:
Thanks--the strong test suite is important to prove behavior for a library that claims to support high-availability. A bug in this package that causes downtime would defeat the purpose of using Sentinel for services designed to endure outages.

Looks like Laravel added queueable closures in version 5.7.7 (as of now, Travis installs 5.7.13). The tests need a tweak in the app double bootstrap routine. If I remember correctly, you're running 5.7.4, so this shouldn't impact your local tests. The 5.7 test error isn't related to this change. I'll push a fix for the tests tonight to the 2.x branch--the PR should pass if you merge it in.